src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketBuildStatusClientImpl.java

```diff
@@ -21,7 +23,10 @@ public class BitbucketBuildStatusClientImpl implements BitbucketBuildStatusClien
     }
 
     @Override
-    public void post(BitbucketBuildStatus buildStatus) {
+    public void post(BitbucketBuildStatus.Builder buildStatusBuilder, Consumer<BitbucketBuildStatus> beforePost) {
+        BitbucketBuildStatus buildStatus = buildStatusBuilder.legacy().build();
```

Comment:
Is this always going to be a legacy build status that's being posted?

Reply:
Yes this is to our old endpoint with no rich build statuses. I'll clarify this